FoodFriends/app/src/main/java/com/example/foodfriends/activities/SettingsActivity.java

```diff
@@ -0,0 +1,91 @@
+package com.example.foodfriends.activities;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.Switch;
+import android.widget.TextView;
+
+import com.example.foodfriends.R;
+import com.example.foodfriends.observable_models.UserObservable;
+import com.parse.ParseUser;
+
+import java.time.temporal.TemporalAccessor;
+import java.util.Observable;
+import java.util.Observer;
+
+public class SettingsActivity extends AppCompatActivity implements Observer {
+    private static final String TAG = "Settings Activity";
+    private UserObservable user;
+    private Button btnLogOut;
+    private EditText etName;
+    private EditText etCity;
+    private EditText etState;
+    private TextView tvDiscard;
+    private TextView tvSave;
+    private Switch switchPrivacy;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_settings);
+        user = getIntent().getParcelableExtra("user");
+        btnLogOut = findViewById(R.id.btnLogOut);
+        etName = findViewById(R.id.etSettingsName);
+        etCity = findViewById(R.id.etSettingsCity);
+        etState = findViewById(R.id.etSettingsState);
+        tvDiscard = findViewById(R.id.tvSettingsDiscard);
+        tvSave = findViewById(R.id.tvSettingsSave);
+        switchPrivacy = findViewById(R.id.switchPrivacy);
+
+        etName.setText(user.getName());
+        etCity.setText(user.getCity());
+        etState.setText(user.getState());
+        switchPrivacy.setChecked(user.getPrivateAccount());
+
+        btnLogOut.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                userLogOut();
+            }
+        });
+        tvDiscard.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                finish();
+            }
+        });
+        tvSave.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                user.setState(etState.getText().toString());
+                user.setCity(etCity.getText().toString());
+                user.setName(etName.getText().toString());
+                user.setPrivateAccount(switchPrivacy.isChecked());
+                user.save_user();
```

Comment:
in a later pull request pls update to camel case

Reply:
updated!